lib/internal/bootstrap/loaders.js

```diff
@@ -208,6 +211,74 @@
         NativeModule.require;
       fn(this.exports, requireFn, this, process);
 
+      if (config.experimentalModules) {
+        if (this.id === 'events') {
+          this.namespace = [];
+        } else {
+          this.namespace = Object.entries(
+            Object.getOwnPropertyDescriptors(this.exports))
+            .filter(([name, d]) => d.enumerable)
+            .map(([name]) => name);
+        }
+
+        const update = (property, value) => {
+          if (this.reflect !== undefined && this.namespace.includes(property))
+            this.reflect.exports[property].set(value);
+        };
+
+        const methodWrapMap = new WeakMap();
+
+        const proxy = new Proxy(this.exports, {
+          set(target, prop, value) {
+            update(prop, value);
+            target[prop] = value;
+            return true;
+          },
+          defineProperty(target, prop, descriptor) {
+            if (this.namespace.includes(prop)) {
+              const valueDescriptor =
+                Reflect.getOwnPropertyDescriptor(descriptor, 'value');
+              if (valueDescriptor &&
+                  Reflect.defineProperty(target, prop, descriptor)) {
+                update(prop, valueDescriptor.value);
+                return true;
+              }
+            }
+            return Reflect.defineProperty(target, prop, descriptor);
+          },
+          deleteProperty(target, prop) {
+            if (this.namespace.includes(prop))
+              return false;
+            return delete target[prop];
+          },
```

Comment:
👆 By returning `false` you're making the CJS export objects non-`delete` able. Instead you could allow the operation to happen, capturing the result of `Reflect.deleteProperty(target, prop)`, calling `update`, then returning the captured result. If a property is deleted its updated value is `undefined` or whatever is exposed on its prototype.

Reply:
> calling update

Reply:
> calling it with what exactly? the property is gone if we let it get deleted

Whatever the value is `target[prop]` value is that's remaining.